### ISSUE TITLE
Incorrect semver when updated to AndroidX

### ISSUE THREAD

**Issue by jonjon1123** [BaseflowIT/flutter-google-api-availability#12]
## 🐛 Bug Report

This library did not follow proper [semver ](https://semver.org/) when it updated to AndroidX. Version 1.0.5 was updated to AndroidX and this is a **breaking change** yet the previous version was 1.0.4!

This could be an even bigger problem if this library is used as a transitive dependency from another library (as it was in my case) as it is not obvious that this library is the issue.

I was able to work around this issue by doing the following in my pubspec.yaml:
```
dependency_overrides:
  google_api_availability: 1.0.4
```

### Expected behavior
I should be able to put `google_api_availability: ^1.0.0` in my pubspec.yaml and rely on that any versions within 1.0.**x** will not break me. This is proper semver and [Dart's versioning](https://www.dartlang.org/tools/pub/dependencies#version-constraints) relies heavily on libraries following semver. The proper version update for the AndroidX compatible release should have been 1.**1**.0. This would have ensured that only people who manully updated to ^1.1.0 would get the breaking change. To see an example of how to do this, look at the [cloud_firestore changelog](https://pub.dartlang.org/packages/cloud_firestore#-changelog-tab-).

### Reproduction steps
Put `google_api_availability: ^1.0.0` in a project that has not been updated to AndroidX. You will get errors when it tries to multidex. Something like:
```
Execution failed for task ':app:transformClassesWithMultidexlistForRelease'.
> com.android.build.api.transform.TransformException: Error while generating the main dex list.
```

### Configuration
A Flutter project that is not updated to AndroidX and uses multidex.

**Version:** 1.0.5

**Platform:** 
- [ ] :iphone: iOS
- [x] :robot: Android

**Comment by mvanbeusekom** [comment]
@jonjon1123 you are a 100% correct this was an error on our side. Thanks for the heads-up though

**Comment by jonjon1123** [comment]
It looks like the there is a way to fix/revert this in Dart pub. Take a look at the [firebase_analytics package changelog](https://pub.dartlang.org/packages/firebase_analytics#-changelog-tab-). They also had an issue with bumping the wrong version number but where able to revert the breaking change on Dart pub. I would highly recommend doing this so more people don't run into this issue.

**Comment by mvanbeusekom** [comment]
Thanks for pointing that out. I have just published the fix 1.0.6+1
